crates/llms/src/embeddings/candle/mod.rs

```diff
@@ -237,7 +239,15 @@ impl Embed for CandleEmbedding {
     }
 
     async fn embed(&self, input: EmbeddingInput) -> Result<Vec<Vec<f32>>> {
-        self._embed(input).map(|(embeddings, _usage)| embeddings)
+        // This needs to be blocking to avoid starving other async tasks in the runtime, since this blocks for long periods of time.
```

Comment:
Is there a reason not to perform the blocking in `Self::_embed`?

Reply:
I chose to keep `Self::_embed` as a non-async function and treat the whole function as blocking